src/main/io/gps.c

```diff
@@ -946,7 +978,9 @@ static void ubloxSetSbas(void)
 void setSatInfoMessageRate(uint8_t divisor)
 {
     // enable satInfoMessage at 1:5 of the nav rate if configurator is connected
-    if (gpsData.ubloxM9orAbove) {
+    if (isRxOnly()) {
+        // can't configure GPS
+    } else if (gpsData.ubloxM9orAbove) {
```

Comment:
empty if block, rely on compiler to remove?
Better do
`if (!isRxOnly() && gpsData.ubloxM9orAbove) {`

Reply:
Compiler will optimize both easily.

But I can't find a way to make the code easy to follow, without refactoring ubloxM?orAbove completely and using `switch`.